core/app/components/FormBuilder/actions.ts

```diff
@@ -16,6 +23,23 @@ import { autoRevalidate } from "~/lib/server/cache/autoRevalidate";
 import { findCommunityBySlug } from "~/lib/server/community";
 import { defineServerAction } from "~/lib/server/defineServerAction";
 
+const upsertRelatedPubTypes = async (
```

Comment:
this is maybe not the most efficient but I think it's the easiest to reason about. let me know if there's a better way to do this though, since this requires a few db calls